### ISSUE TITLE
Escape from nav item and footer titles

### ISSUE THREAD

**Issue by Baltazardoung** [squidfunk/mkdocs-material#1961]
Page titles may contain double quotes. But in navigations and footers (and maybe elsewhere), they are not escaped.

```html
  <li class="{{ class }}">
    <a href="{{ nav_item.url | url }}" title="{{ nav_item.title | striptags | escape }}" class="md-nav__link">
      {{ nav_item.title }}
    </a>
  </li>
```
What produces:

```
    <a href="revert.html" title="Revenir en arrière (*aka.* "faire un revert")" class="md-nav__link md-nav__link--active">
      Revenir en arrière (*aka.* "faire un revert")
    </a>
```

![image](https://user-images.githubusercontent.com/13434663/94923823-c75c2680-04bc-11eb-92ef-53f3ac8d5d1b.png)

![image](https://user-images.githubusercontent.com/13434663/94923842-cd520780-04bc-11eb-9300-6c9ca9ce6afe.png)

**Suggestion**
Add escape to the item titles  (and others attributs) in `nav-item.html` and `footer.html` :

    <a href="{{ nav_item.url | url }}" title="{{ nav_item.title | striptags | escape }}" class="md-nav__link">
      {{ nav_item.title }}
    </a>

**Comment by squidfunk** [comment]
We already started a PR for this in https://github.com/squidfunk/mkdocs-material/pull/1117, but it wasn't merged. You may create a new PR, and we can start collaborating.

**Comment by squidfunk** [comment]
I've had some time to revisit this topic. In turns out that most of the `title` links can be removed, as they duplicate the anchor text. This is true for the navigation item links and footer navigation links. This effort was started in https://github.com/squidfunk/mkdocs-material/issues/1207 and https://github.com/squidfunk/mkdocs-material/pull/1240, but it seems we forgot about other places in the template. 52787c6a removes those attributes.

The only remaining, user-definable title attributes are the [social links](https://squidfunk.github.io/mkdocs-material/setup/adding-social-links/#name) and site name, which are now properly escaped as of e149006a

**Comment by squidfunk** [comment]
Released as part of 6.0.2

**Comment by Baltazardoung** [comment]
Thank you very much for that.

In our case it is important that the html code is not broken as we are building the documentation from several projects included as a git submodules and using a CI (with html-proofer) to verify that the internal links and others are not broken. And without the escaped titles, html-proofer generates a fatal error because the source HTML code is not valid.
